Sources/StreamChatUI/Utils/Debouncer.swift

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+public protocol Debouncing {
```

Comment:
I think we already had some logic with debouncing, @nuno-vieira?

Reply:
We don't have it implemented yet, it was proposed here: https://github.com/GetStream/ios-issues-tracking/issues/217. Debouncing has some gotchas as far as I remember, worth comparing the implementations 👍

Reply:
Comparing the implementation with what?

Reply:
Nevermind 👍